sdcm/cluster.py

```diff
@@ -2411,12 +2411,20 @@ def start_scylla(self, verify_up=True, verify_down=False, timeout=500):
         if verify_up:
             self.wait_jmx_up(timeout=timeout)
 
-    @retrying(n=3, sleep_time=5, allowed_exceptions=NETWORK_EXCEPTIONS + (CommandTimedOut, ),
+    @retrying(n=3, sleep_time=5, allowed_exceptions=NETWORK_EXCEPTIONS,
               message="Failed to stop scylla.server, retrying...")
     def stop_scylla_server(self, verify_up=False, verify_down=True, timeout=300, ignore_status=False):
         if verify_up:
             self.wait_db_up(timeout=timeout)
-        self.stop_service(service_name='scylla-server', timeout=timeout, ignore_status=ignore_status)
+        try:
+            self.stop_service(service_name='scylla-server', timeout=1, ignore_status=ignore_status)
+        except CommandTimedOut:
+            LOGGER.error(
+                "timeout during stopping scylla-server, collect diagnostic data and wait for SIGABRT triggered by systemd")
+            collect_diagnostic_data(self)
+            wait.wait_for(func=lambda: "Active: inactive" in
```

Comment:
Isn't this line doing the same as the below `verify_down` ?

Reply:
it checks `is_port_used` - and in case of freeze after closing port, we don't know - I wanted to simulate same behavior as systemctl - waiting for status inactive. I'm not sure about this solution thou.